Parallel upload of files to girder

subdavis opened the pull request:
Still testing, but this is more or less done.

closes #231

subdavis commented on src/utils/mixins.js:
```diff
@@ -183,60 +186,64 @@ const fileUploader = {
+              file.status = 'error';
+              this.uploading = false;
+              this.$emit('error', { error, file });
+              throw error;
```
This is behavior is different from what was here before.  Now, unless the caller of `start()` handles errors, this will be uncaught.   I tend to prefer that unhandled errors don't get swallowed by this library.

It will also be uncaught in Upload.vue.  I don't really see that as a problem, but we can catch it there if necessary.

zachmullen replied:
I like the idea of throwing here, however, shouldn't we handle it in Upload.vue since that's where `start` is called? Otherwise wouldn't the only way to catch an exception during upload be to manually call `startUpload` from the outside?

subdavis replied:
> shouldn't we handle it in Upload.vue since that's where start is called

Sure, an empty handler can be added to Upload.vue.

zachmullen replied:
Sorry, I may be misunderstanding something. Prior to this change, my understanding was that the Upload.vue *did* handle the errors, by presenting the error message, and allowing "Resume" to be clicked. Was that not the case?

subdavis replied:
No.  prior to this change, `start()` swallowed the error and expected the mixin consumer to present information about the error using `this.errorMessage`.  `errorMessage` is useful, but I also think `start()` should throw the exception too in case some external mixin consumer has other handling they want to do.

subdavis replied:
I should say `start()` swallowed _some_ of the errors.  preUpload and postUpload were unhandled, but beforeUpload, afterUpload, and all axios errors were handled.

zachmullen commented on src/utils/mixins.js:
```diff
@@ -183,60 +186,64 @@ const fileUploader = {
-      for (; this.currentIndex < this.files.length; this.currentIndex += 1) {
-        const file = this.files[this.currentIndex];
+      this.indeterminate = false;
+      for (let i = 0; i < this.files.length; i += 1) {
```
This looks like it could be replaced with `const promiseList = this.files.map(...)`

mgrauer replied:
I'm pretty sure `const promise list` is the name of a love song.

zachmullen commented:
Do we know why CI isn't reporting?

subdavis commented:
@zachmullen think it was just related to outages yesterday https://www.githubstatus.com/.  Webhooks were down, so CI might not have been reported.

I've squashed and added a release version.  Could you review again?  TY
